internal/pkg/aws/sessions/sessions.go

```diff
@@ -70,6 +70,14 @@ func UserAgentExtras(extras ...string) func(*Provider) {
 	}
 }
 
+// UserAgentExtras adds additional User-Agent extras to cached sessions and any new sessions.
+func (p *Provider) UserAgentExtras(extras ...string) {
```

Comment:
What extra things happens here in addition to existing `UserAgentExtras` function?

Reply:
The `UserAgentExtras` function can only be used in the constructor, the method is so that you can add any extras at any point.

So we can start with a provider that's like this for a command:

```
provider := sessions.NewProvider(sessions.UserAgentExtras("svc package"))

....
if override.IsCDK() {
   provider.UserAgentExtras("cdk override")
}
```